test/k8sT/Istio.go

```diff
@@ -0,0 +1,171 @@
+		istioConfigMap   string = helpers.ManifestGet("istio-kube-inject-template.yaml")
+		istioInitPolicy  string = helpers.ManifestGet("istio-sidecar-init-policy.yaml")
+		demoPath         string = helpers.ManifestGet("demo_big.yaml")
+		container        string = "app-frontend"
+	)
+
+	BeforeAll(func() {
+		logger = log.WithFields(logrus.Fields{"testName": "K8sValidatedIstioTest"})
+		logger.Info("Starting")
+
+		kubectl = helpers.CreateKubectl(helpers.K8s1VMName(), logger)
+		kubectl.Apply(ciliumPath)
+
+		_ = kubectl.Exec(fmt.Sprintf(
+			"%s -n %s delete pods %s",
+			helpers.KubectlCmd, helpers.KubeSystemNamespace, ciliumFilter))
+
+		err := kubectl.WaitforPods(helpers.KubeSystemNamespace, ciliumFilter, 600)
+		Expect(err).Should(BeNil())
+
+		err = kubectl.WaitKubeDNS()
+		Expect(err).Should(BeNil(), "DNS is not ready after timeout")
+
+		_, err = kubectl.CiliumPolicyAction(
+			helpers.KubeSystemNamespace, istioInitPolicy,
+			helpers.KubectlApply, 300)
+		Expect(err).Should(BeNil(), "Cannot apply init policy")
+
+		ciliumPodK8s1, err = kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
+		Expect(err).Should(BeNil(), "Cannot get cilium pod on k8s1")
+
+		// Installing Istio
+		res := kubectl.Exec(fmt.Sprintf("cd /tmp/; curl -L %s | tar xz ", istioURL))
+		res.ExpectSuccess("Cannot download istio: %s", res.CombineOutput())
+
+		res = kubectl.ExecWithSudo(fmt.Sprintf("mv -f /tmp/istio-%s /usr/local/istio", istioVersion))
+		res.ExpectSuccess("Cannot mv Istio source folder ")
+
+		res = kubectl.ExecWithSudo("ln -sf /usr/local/istio/bin/* /usr/local/bin/")
+		res.ExpectSuccess("cannot copy Istio binaries")
+
+		kubectl.Apply(istioManifest)
+
+		err = kubectl.WaitforPods(istioNS, istioFilter, 600)
+		Expect(err).Should(BeNil(), "Istio cannot be installed correctly")
+	})
+
+	kubeInject := func(fileName string, command string) *helpers.CmdRes {
+		cmd := fmt.Sprintf("istioctl kube-inject --injectConfigFile %s -f %s | %s %s -f -",
+			istioConfigMap,
+			fileName,
+			helpers.KubectlCmd,
+			command)
+		return kubectl.Exec(cmd)
+	}
+
+	AfterAll(func() {
+		kubectl.Delete(ciliumPath)
+		kubectl.Delete(istioManifest)
+		kubectl.ExecWithSudo("rm -rf /ust/local/istio")
+	})
+
+	AfterFailed(func() {
+		kubectl.CiliumReport(helpers.KubeSystemNamespace,
+			"cilium service list",
+			"cilium policy get",
+			"cilium endpoint list")
+	})
+
+	JustBeforeEach(func() {
+		microscopeErr, microscopeCancel = kubectl.MicroscopeStart()
+		Expect(microscopeErr).To(BeNil(), "Microscope cannot be started")
+	})
+
+	JustAfterEach(func() {
+		kubectl.ValidateNoErrorsOnLogs(CurrentGinkgoTestDescription().Duration)
+		Expect(microscopeCancel()).To(BeNil(), "cannot stop microscope")
+	})
+
+	AfterEach(func() {
+		err := kubectl.WaitCleanAllTerminatingPods()
+		Expect(err).To(BeNil(), "Terminating containers are not deleted after timeout")
+	})
+
+	Context("With Basic Application", func() {
```

Comment:
What is a 'Basic Application' and a 'Basic Connectivity Test'? I understand that the description of a Context / It is brief in nature, but this provides no added value to indicate what is actually being tested here. A brief comment would be helpful indicating what exactly is being tested here, or better yet, indicate what type of datapath traffic is being tested, etc.